extensions/standard-processors/processors/AppendHostInfo.cpp

```diff
@@ -23,90 +23,92 @@
+  else
+    interface_name_filter_ = utils::nullopt;
+  std::string refresh_policy;
+  context->getProperty(RefreshPolicy.getName(), refresh_policy);
+  if (refresh_policy == REFRESH_POLICY_ON_TRIGGER)
+    refresh_on_trigger_ = true;
+
+  if (!refresh_on_trigger_)
+    refreshHostInfo();
 }
 
-void AppendHostInfo::onTrigger(core::ProcessContext *context, core::ProcessSession *session) {
+void AppendHostInfo::onTrigger(core::ProcessContext*, core::ProcessSession* session) {
   std::shared_ptr<core::FlowFile> flow = session->get();
   if (!flow)
     return;
 
-  // Get Hostname
-
-  std::string hostAttribute = "";
-  context->getProperty(HostAttribute.getName(), hostAttribute);
-  flow->addAttribute(hostAttribute.c_str(), org::apache::nifi::minifi::io::Socket::getMyHostName());
+  if (refresh_on_trigger_)
+    refreshHostInfo();
 
-  // Get IP address for the specified interface
-  std::string iface;
-  context->getProperty(InterfaceName.getName(), iface);
-  // Confirm the specified interface name exists on this device
-#ifndef WIN32
-  if (if_nametoindex(iface.c_str()) != 0) {
-    struct ifreq ifr;
-    int fd = socket(AF_INET, SOCK_DGRAM, 0);
-    // Type of address to retrieve - IPv4 IP address
-    ifr.ifr_addr.sa_family = AF_INET;
-    // Copy the interface name in the ifreq structure
-    strncpy(ifr.ifr_name, iface.c_str(), IFNAMSIZ - 1);
-    ioctl(fd, SIOCGIFADDR, &ifr);
-    close(fd);
-
-    std::string ipAttribute;
-    context->getProperty(IPAttribute.getName(), ipAttribute);
-    flow->addAttribute(ipAttribute.c_str(), inet_ntoa(((struct sockaddr_in *) &ifr.ifr_addr)->sin_addr));
+  flow->addAttribute(hostname_attribute_name_, hostname_);
+  if (ipaddresses_.has_value()) {
+    flow->addAttribute(ipaddress_attribute_name_, ipaddresses_.value());
   }
-#endif
 
-  // Transfer to the relationship
   session->transfer(flow, Success);
 }
 
+void AppendHostInfo::refreshHostInfo() {
+  hostname_ = org::apache::nifi::minifi::io::Socket::getMyHostName();
+  auto filter = [this](const utils::NetworkInterfaceInfo& interface_info) -> bool {
+    bool has_ipv4_address = interface_info.hasIpV4Address();
+    bool matches_regex_or_empty_regex = (!interface_name_filter_.has_value()) || std::regex_match(interface_info.getName(), interface_name_filter_.value());
+    return has_ipv4_address && matches_regex_or_empty_regex;
+  };
+  auto network_interface_infos = utils::NetworkInterfaceInfo::getNetworkInterfaceInfos(filter);
+  std::ostringstream oss;
+  if (network_interface_infos.size() == 0) {
+    ipaddresses_ = utils::nullopt;
+  } else {
+    for (auto& network_interface_info : network_interface_infos) {
+      auto& ip_v4_addresses = network_interface_info.getIpV4Addresses();
+      std::copy(std::begin(ip_v4_addresses), std::end(ip_v4_addresses), std::ostream_iterator<std::string>(oss, ","));
+    }
+    ipaddresses_ = oss.str();
+    ipaddresses_.value().pop_back();  // to remove trailing comma
```

Comment:
That couldnt happen because there is a filter before this which makes sure that only network interface infos with ipv4 addresses are returned.  `has_ipv4_address && matches_regex_or_empty_regex;`